### PR TITLE
Fix overwriting accessors queue

### PR DESCRIPTION
**Pull request by kustosz**
This fixes a rather nasty bug that caused the typechecker to forget to resolve some assumptions. Any carry-over method assumption appearing during method importing was immediately dropped from the solver queue.

### CODE REVIEW

File: passes/src/Luna/Pass/Resolution/MethodResolution.hs [hunk 1]
```diff
@@ -56,12 +57,16 @@ runMethodResolution :: MonadPassManager m => SubPass AccessorFunction m Bool
 runMethodResolution = do
     accs <- getAccs <$> getAttr @UnresolvedAccs
     res <- forM accs $ \acc -> do
-        result <- importAccessor acc
         case result of
-            False -> return $ Just acc
-            _     -> return Nothing
-    putAttr @UnresolvedAccs $ putAccs $ catMaybes res
-    return $ any isNothing res
+            True -> return $ Just acc
```
Comment (piotrMocz) [inline passes/src/Luna/Pass/Resolution/MethodResolution.hs]: Where does `result` come from? Does the code compile?

File: passes/src/Luna/Pass/Resolution/MethodResolution.hs [hunk 2]
```diff
@@ -56,12 +57,16 @@ runMethodResolution :: MonadPassManager m => SubPass AccessorFunction m Bool
 runMethodResolution = do
     accs <- getAccs <$> getAttr @UnresolvedAccs
     res <- forM accs $ \acc -> do
-        result <- importAccessor acc
```
Comment (piotrMocz) [inline passes/src/Luna/Pass/Resolution/MethodResolution.hs]: This line probably shouldn't have been removed :)

File: passes/src/Luna/Pass/Resolution/MethodResolution.hs (hunk 1 above)
Comment (kustosz) [reply passes/src/Luna/Pass/Resolution/MethodResolution.hs]: 2 lines above, WDYM?

Comment (piotrMocz) [reply passes/src/Luna/Pass/Resolution/MethodResolution.hs]: Didn't know you can push amended commits and github doesn't mind :(

Comment (kustosz) [reply passes/src/Luna/Pass/Resolution/MethodResolution.hs]: It's a feature, and a pretty good one I'd say.

**Pull request merged**